client/remoterepo.c

```diff
@@ -748,11 +747,13 @@ TDNFParseAndGetURLFromMetalink(
             dwError = ERROR_TDNF_METALINK_RESOURCE_VALIDATION_FAILED;
             BAIL_ON_TDNF_ERROR(dwError);
         }
-        strncpy(pszBaseURLFromMetalink, (*resources)->url, strlen((*resources)->url));
-        dwError = TDNFTrimSuffix(pszBaseURLFromMetalink, TDNF_REPO_METADATA_FILE_PATH);
+
+        strncpy(buf, (*resources)->url, BUFSIZ-1);
+        buf[BUFSIZ-1] = '\0'; // force terminate
```

Comment:
This is not needed as buf is already initialised with zero and we are copying only BUFSIZ-1 len of chars